pkg/ciliumenvoyconfig/cec_manager.go

```diff
@@ -271,8 +271,8 @@ func (r *cecManager) getEndpoint(serviceName string, serviceNamespace string) (*
 	var res *k8s.Endpoints
 	iter := store.IterKeys()
 	for iter.Next() {
-		e, _, _ := store.GetByKey(iter.Key())
-		if e.EndpointSliceID.ServiceID.Name == serviceName &&
+		e, exists, error := store.GetByKey(iter.Key())
+		if exists && error == nil && e.EndpointSliceID.ServiceID.Name == serviceName &&
```

Comment:
Shouldn't `if exists && e.EndpointSliceID.ServiceID.Name == serviceName &&` be sufficient?

Reply:
If we can assert that `exists -> !error` for all inputs, then yes, but I don't know enough about current (or future) underlying implementations to make that assertion. Happy to change this if someone smarter than me can make that assertion. 😄